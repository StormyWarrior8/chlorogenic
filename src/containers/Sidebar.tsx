import CircleEditOutlineIcon from "mdi-react/CircleEditOutlineIcon"
import React from "react"
import { ColumnPanel } from "../models/columns"
import { Button, Flexbox } from "../UX"
import styled from "../UX/Styled"

interface SidebarProps {
  panelIndex: number
  columns: ColumnPanel[]
  onClick?(index: number): void
  onClickAdd?(): void
  onClickEdit?(index: number): void
}
const SideBar = styled(Flexbox)`
  flex-direction: column;
  width: 12em;
`
const UL = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
`
const LI = styled.li<{ active: boolean }>`
  padding-left: 1em;
  text-indent: -0.7em;

  &::before {
    content: "â€¢";
    color: ${({ theme, active }) =>
      active ? theme.greenColor : theme.baseColor};
    margin-right: 0.2em;
  }
  > span {
    cursor: pointer;
  }
`

const View = ({
  panelIndex,
  columns,
  onClick,
  onClickAdd,
  onClickEdit,
}: SidebarProps) => {
  return (
    <SideBar>
      <UL>
        {columns.map((c, i) => (
          <LI active={panelIndex === i} key={i}>
            <span onClick={() => onClick && onClick(i)}>
              {c.name || `${i}-No name`}{" "}
            </span>
            <span onClick={() => onClickEdit && onClickEdit(i)}>
              {" "}
              <CircleEditOutlineIcon size={14} />
            </span>
          </LI>
        ))}
      </UL>
      <Button onClick={onClickAdd}>Add new panel</Button>
    </SideBar>
  )
}

export default View
