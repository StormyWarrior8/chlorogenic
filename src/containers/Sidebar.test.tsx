import { navigate } from "@reach/router"
import * as React from "react"
import * as renderer from "react-test-renderer"
import Sidebar from "./Sidebar"

it("should change active className", () => {
  const panels = [
    {
      name: "hoge",
      columns: [],
    },
    {
      name: "huga",
      columns: [],
    },
  ]

  panels.forEach((_, index) => {
    // if navigated to "/0", the "/0" link becomes .nav-active. the others are .nav.
    // if navigated to "/1", the "/1" link becomes .nav-active. the others are .nav.

    navigate(`/${index}`)
    const tree = renderer
      .create(<Sidebar panelIndex={index} panels={panels} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
