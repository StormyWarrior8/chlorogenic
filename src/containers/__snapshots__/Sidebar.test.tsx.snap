// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change active className 1`] = `
<div
  class="sc-ifAKCX mOEEI sc-bxivhb glCqXO"
>
  <ul
    class="sc-EHOje gMzKpB"
  >
    <li
      class="sc-bZQynM ihCmRw"
    >
      <a
        class="nav"
        href="/0"
      >
        hoge
      </a>
      <span>
         
        <svg
          class="Icon"
          style="width: 0.8999999999999999rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12H20C20,16.42 16.42,20 12,20C7.58,20 4,16.42 4,12C4,7.58 7.58,4 12,4V2M18.78,3C18.61,3 18.43,3.07 18.3,3.2L17.08,4.41L19.58,6.91L20.8,5.7C21.06,5.44 21.06,5 20.8,4.75L19.25,3.2C19.12,3.07 18.95,3 18.78,3M16.37,5.12L9,12.5V15H11.5L18.87,7.62L16.37,5.12Z"
            style="fill: currentColor;"
          />
        </svg>
      </span>
    </li>
    <li
      class="sc-bZQynM ihCmRw"
    >
      <a
        class="nav"
        href="/1"
      >
        huga
      </a>
      <span>
         
        <svg
          class="Icon"
          style="width: 0.8999999999999999rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12H20C20,16.42 16.42,20 12,20C7.58,20 4,16.42 4,12C4,7.58 7.58,4 12,4V2M18.78,3C18.61,3 18.43,3.07 18.3,3.2L17.08,4.41L19.58,6.91L20.8,5.7C21.06,5.44 21.06,5 20.8,4.75L19.25,3.2C19.12,3.07 18.95,3 18.78,3M16.37,5.12L9,12.5V15H11.5L18.87,7.62L16.37,5.12Z"
            style="fill: currentColor;"
          />
        </svg>
      </span>
    </li>
  </ul>
  <button
    class="sc-bdVaJa dnqbXl"
  >
    Add new panel
  </button>
  <a
    href="https://github.com/mironal/chlorogenic"
    target="_blank"
  >
    <svg
      class="Icon"
      style="width: 0.8999999999999999rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.38,8.53C20.54,8.13 21.06,6.54 20.21,4.39C20.21,4.39 18.9,4 15.91,6C14.66,5.67 13.33,5.62 12,5.62C10.68,5.62 9.34,5.67 8.09,6C5.1,3.97 3.79,4.39 3.79,4.39C2.94,6.54 3.46,8.13 3.63,8.53C2.61,9.62 2,11 2,12.72C2,19.16 6.16,20.61 12,20.61C17.79,20.61 22,19.16 22,12.72C22,11 21.39,9.62 20.38,8.53M12,19.38C7.88,19.38 4.53,19.19 4.53,15.19C4.53,14.24 5,13.34 5.8,12.61C7.14,11.38 9.43,12.03 12,12.03C14.59,12.03 16.85,11.38 18.2,12.61C19,13.34 19.5,14.23 19.5,15.19C19.5,19.18 16.13,19.38 12,19.38M8.86,13.12C8.04,13.12 7.36,14.12 7.36,15.34C7.36,16.57 8.04,17.58 8.86,17.58C9.69,17.58 10.36,16.58 10.36,15.34C10.36,14.11 9.69,13.12 8.86,13.12M15.14,13.12C14.31,13.12 13.64,14.11 13.64,15.34C13.64,16.58 14.31,17.58 15.14,17.58C15.96,17.58 16.64,16.58 16.64,15.34C16.64,14.11 16,13.12 15.14,13.12Z"
        style="fill: currentColor;"
      />
    </svg>
  </a>
</div>
`;

exports[`should change active className 2`] = `
<div
  class="sc-ifAKCX mOEEI sc-bxivhb glCqXO"
>
  <ul
    class="sc-EHOje gMzKpB"
  >
    <li
      class="sc-bZQynM ihCmRw"
    >
      <a
        aria-current="page"
        class="nav-active"
        href="/0"
      >
        hoge
      </a>
      <span>
         
        <svg
          class="Icon"
          style="width: 0.8999999999999999rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12H20C20,16.42 16.42,20 12,20C7.58,20 4,16.42 4,12C4,7.58 7.58,4 12,4V2M18.78,3C18.61,3 18.43,3.07 18.3,3.2L17.08,4.41L19.58,6.91L20.8,5.7C21.06,5.44 21.06,5 20.8,4.75L19.25,3.2C19.12,3.07 18.95,3 18.78,3M16.37,5.12L9,12.5V15H11.5L18.87,7.62L16.37,5.12Z"
            style="fill: currentColor;"
          />
        </svg>
      </span>
    </li>
    <li
      class="sc-bZQynM ihCmRw"
    >
      <a
        class="nav"
        href="/1"
      >
        huga
      </a>
      <span>
         
        <svg
          class="Icon"
          style="width: 0.8999999999999999rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,2C6.48,2 2,6.48 2,12C2,17.52 6.48,22 12,22C17.52,22 22,17.52 22,12H20C20,16.42 16.42,20 12,20C7.58,20 4,16.42 4,12C4,7.58 7.58,4 12,4V2M18.78,3C18.61,3 18.43,3.07 18.3,3.2L17.08,4.41L19.58,6.91L20.8,5.7C21.06,5.44 21.06,5 20.8,4.75L19.25,3.2C19.12,3.07 18.95,3 18.78,3M16.37,5.12L9,12.5V15H11.5L18.87,7.62L16.37,5.12Z"
            style="fill: currentColor;"
          />
        </svg>
      </span>
    </li>
  </ul>
  <button
    class="sc-bdVaJa dnqbXl"
  >
    Add new panel
  </button>
  <a
    href="https://github.com/mironal/chlorogenic"
    target="_blank"
  >
    <svg
      class="Icon"
      style="width: 0.8999999999999999rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.38,8.53C20.54,8.13 21.06,6.54 20.21,4.39C20.21,4.39 18.9,4 15.91,6C14.66,5.67 13.33,5.62 12,5.62C10.68,5.62 9.34,5.67 8.09,6C5.1,3.97 3.79,4.39 3.79,4.39C2.94,6.54 3.46,8.13 3.63,8.53C2.61,9.62 2,11 2,12.72C2,19.16 6.16,20.61 12,20.61C17.79,20.61 22,19.16 22,12.72C22,11 21.39,9.62 20.38,8.53M12,19.38C7.88,19.38 4.53,19.19 4.53,15.19C4.53,14.24 5,13.34 5.8,12.61C7.14,11.38 9.43,12.03 12,12.03C14.59,12.03 16.85,11.38 18.2,12.61C19,13.34 19.5,14.23 19.5,15.19C19.5,19.18 16.13,19.38 12,19.38M8.86,13.12C8.04,13.12 7.36,14.12 7.36,15.34C7.36,16.57 8.04,17.58 8.86,17.58C9.69,17.58 10.36,16.58 10.36,15.34C10.36,14.11 9.69,13.12 8.86,13.12M15.14,13.12C14.31,13.12 13.64,14.11 13.64,15.34C13.64,16.58 14.31,17.58 15.14,17.58C15.96,17.58 16.64,16.58 16.64,15.34C16.64,14.11 16,13.12 15.14,13.12Z"
        style="fill: currentColor;"
      />
    </svg>
  </a>
</div>
`;
